### PR TITLE
Louis/stap

### PR DESCRIPTION
**Pull request by LinuxGit**
- add systemtap INSTALL doc
- add iotrace and context switch script

### CODE REVIEW

**Comment by LinuxGit** [comment]
@siddontang  PTAL

File: systemtap/cstop_pid.stp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+probe scheduler.cpu_off {
+  csw_count[task_prev, task_next]++
+  idle_count+=idle
+}
+
+function fmt_task(task_prev, task_next)
+{
+   return sprintf("%s(%d)->%s(%d)",
+                                task_execname(task_prev),
```
Comment (siddontang) [inline systemtap/cstop_pid.stp]: make alignment

File: systemtap/cstop_pid.stp [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#! /usr/bin/env stap
+# Usage: sudo stap cs_trace_pid.stp <interval>
+# Example: sudo stap cs_trace_pid.stp 1
+
+global csw_count[999999]
+global idle_count
+
+probe scheduler.cpu_off {
```
Comment (siddontang) [inline systemtap/cstop_pid.stp]: do we need to echo a header in the begin function?

File: systemtap/iotrace.stp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+probe begin
+{
+	printf("begin to trace thread IO");
+}
+
+probe vfs.read
+{
+	if (target() == pid()) {
+		printf("%s[%ld] - read %d from %s\n", execname(), tid(), bytes_to_read, reverse_path_walk($file->f_path->dentry))
```
Comment (siddontang) [inline systemtap/iotrace.stp]: seem bytes_to_read is not the real data read from IO, maybe we should use vfs.read.return to get the return size?

File: systemtap/sched_switch.stp [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+ * + indicates that prev process has woken up the next process.
+ * The usage is sched_switch.stp <"pid"/"name"> pid/name
+ */
+
+global target_pid
+global target_name
+
+function state_calc(state) {
+	if(state == 0)
+	status = "R"
```
Comment (siddontang) [inline systemtap/sched_switch.stp]: make alignment

File: systemtap/sched_switch.stp [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+ * + indicates that prev process has woken up the next process.
+ * The usage is sched_switch.stp <"pid"/"name"> pid/name
+ */
+
+global target_pid
+global target_name
+
+function state_calc(state) {
+    if(state == 0)
+    status = "R"
```
Comment (siddontang) [inline systemtap/sched_switch.stp]:
```
if(state == 0)
    status = "R"
if()
```

File: systemtap/sched_switch.stp [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        log("Wrong number of arguments, use none, 'pid nr' or 'name proc'")
+        exit()
+    %)
+
+    %( $# == 2 %?
+        if(@1 == "pid")
+            target_pid = strtol(@2, 10)
+        if(@1 == "name")
+            target_name = @2
+    %)
```
Comment (siddontang) [inline systemtap/sched_switch.stp:94]: can you output the field header name here?

Comment (LinuxGit) [reply systemtap/sched_switch.stp:94]: No problem. PTAL.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**